### PR TITLE
[RNG] Implement Cauchy and extreme value distributions

### PR DESCRIPTION
**Pull request by kcenia4010**
Implement Cauchy and extreme value distributions and  test

### CODE REVIEW

**Review by nsemin — APPROVED** [review]

File: include/oneapi/dpl/internal/random_impl/extreme_value_distribution.h [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        return generate_vec<_Ndistr, _Engine>(__engine, __params);
+    }
+
+    // Specialization of the scalar generation
+    template <int _Ndistr, class _Engine>
+    typename ::std::enable_if<(_Ndistr == 0), result_type>::type
+    generate(_Engine& __engine, const param_type& __params)
+    {
+        oneapi::dpl::exponential_distribution<scalar_type> __e;
+        return __params.a - __params.b * sycl::log(__e(__engine));
```
Comment (aelizaro) [inline include/oneapi/dpl/internal/random_impl/extreme_value_distribution.h:147]: could we have log(0) here?

Comment (kcenia4010) [reply include/oneapi/dpl/internal/random_impl/extreme_value_distribution.h]: It possible as I see. I added the callback.

File: include/oneapi/dpl/internal/random_impl/cauchy_distribution.h [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    param(const param_type& __param)
+    {
+        a_ = __param.a;
+        b_ = __param.b;
+    }
+
+    scalar_type
+    min() const
+    {
+        return -std::numeric_limits<scalar_type>::max();
```
Comment (paveldyakov) [inline include/oneapi/dpl/internal/random_impl/cauchy_distribution.h:83]: Can we use lowest() here?

Comment (kcenia4010) [reply include/oneapi/dpl/internal/random_impl/cauchy_distribution.h]: done

**Review by paveldyakov — APPROVED** [review]

**Review by nsemin — APPROVED** [review]

**Review by aelizaro — APPROVED** [review]

**Pull request merged**